### ISSUE TITLE
AttributeError: 'NoneType' object has no attribute 'shape'

### ISSUE THREAD

**Issue by kaytiii** [m-ama/PyDesigner#63]
$ python pydesigner.py --denoise --degibbs --smooth --rician file.nii -o outfolder
Given DWI is partial fourier, overriding --degibbs; no unringing correction will be done to avoid artifacts.
Path: /Users/kayti/Desktop/Projects/DESIGNER/KTGFDesignerTest/Data/5001/designer/DIFF_siemens_dir_2_7_AVE2
Name: rsdDIFF_siemens_dir_2_7_AVE2
Full: /Users/kayti/Desktop/Projects/DESIGNER/KTGFDesignerTest/Data/5001/designer/DIFF_siemens_dir_2_7_AVE2/rsdDIFF_siemens_dir_2_7_AVE2.nii
No brain mask supplied
Image preprocessed_dwi.nii loaded successfully
Processing with 16 workers...
IRLLS: Noise Estimation : 100%|█| 329476/329476 [00:15<00:00, 21219.59vox/s]
IRLLS: Outlier Detection: 100%|█| 329476/329476 [01:24<00:00, 3907.48vox/s]
Constrained Tensor Fit  : 100%|█| 329476/329476 [04:31<00:00, 1214.13vox/s]
DTI params              : 100%|█| 329476/329476 [00:22<00:00, 14779.67vox/s]
Traceback (most recent call last):
  File "/Users/kayti/Repos/PyDesigner/designer/pydesigner.py", line 560, in <module>
    dp.writeNii(exec(f), img.hdr, fpath)
  File "/Users/kayti/Repos/PyDesigner/designer/fitting/dwipi.py", line 1803, in writeNii
    clipped_img = nib.Nifti1Image(map, hdr.affine, hdr.header)
  File "/Users/kayti/anaconda3/lib/python3.7/site-packages/nibabel/nifti1.py", line 1774, in __init__
    file_map)
  File "/Users/kayti/anaconda3/lib/python3.7/site-packages/nibabel/analyze.py", line 923, in __init__
    dataobj, affine, header, extra, file_map)
  File "/Users/kayti/anaconda3/lib/python3.7/site-packages/nibabel/spatialimages.py", line 468, in __init__
    self.update_header()
  File "/Users/kayti/anaconda3/lib/python3.7/site-packages/nibabel/nifti1.py", line 2046, in update_header
    super(Nifti1Image, self).update_header()
  File "/Users/kayti/anaconda3/lib/python3.7/site-packages/nibabel/nifti1.py", line 1809, in update_header
    super(Nifti1Pair, self).update_header()
  File "/Users/kayti/anaconda3/lib/python3.7/site-packages/nibabel/spatialimages.py", line 490, in update_header
    shape = self._dataobj.shape
AttributeError: 'NoneType' object has no attribute 'shape'

**Comment by jbteves** [comment]
What does the temp directory have in it? I wonder if there's a file missing.

**Comment by TheJaeger** [comment]
I am getting the same error after PR merge. I have two comments after further examination:

1. Functions fails at the for loop in [L646-648](https://github.com/m-ama/PyDesigner/blob/f08d765b58218137ce2d42b33935cbba8eda56b4/designer/pydesigner.py#L646-L648) when converting string to system variable.

2. [Line 659](https://github.com/m-ama/PyDesigner/blob/f08d765b58218137ce2d42b33935cbba8eda56b4/designer/pydesigner.py#L659) of PyDesiner.py is using the wrong parameters for `dwipi.writeNii`. Presently both `map=output_path` and `outdir=output_path`, which is incorrect. Refer to the docstring below for correct parameters:
```
Parameters
    ----------
    map:    3D matrix to write
    header: NifTi header file containing NifTi properties
    outDir: string
            Output file name
    range:  Range to clip images, inclusive of values in range
```

**Comment by jbteves** [comment]
Okay, on further examination it looks like `trace` is failing specifically, which means that the `trace` return value of `img.extractDTI` is `None` for some reason. Let me see if I can dig up what's going on.

**Comment by TheJaeger** [comment]
> Okay, on further examination it looks like `trace` is failing specifically, which means that the `trace` return value of `img.extractDTI` is `None` for some reason. Let me see if I can dig up what's going on.

@jbteves funny. Maybe make a logic for `trace` where it won't save the trace from DTI `if img.isDKI() is True`.

**Comment by vadisetiyo** [comment]
I also received this same error today

**Comment by TheJaeger** [comment]
Reverted PR #76 with commit ba7b69b1e7c00e5bc2876bf92390fc28529f9e53. We're investigating the issue right now.

**Comment by TheJaeger** [comment]
Fixed bt PR #80
